syncopy/tests/synth_data.py

```diff
@@ -202,29 +233,27 @@ def AR2_network(AdjMat=None, nSamples=1000, alphas=[0.55, -0.8]):
 
     sol = np.zeros((nSamples, nChannels))
     # pick the 1st values at random
-    sol[:2, :] = np.random.randn(2, nChannels)
+    rng = np.random.default_rng(seed)
```

Comment:
Just wondering, there is no way to solve that more generally? However something like a global `rng` can't be it, as then the order of calls destroys reproducibility?

Reply:
Yes, using the global RNG is not considered good practice anymore according to the (now accepted) [NEP 19](https://numpy.org/neps/nep-0019-rng-policy.html). Quoting :

_The preferred best practice for getting reproducible pseudorandom numbers is to instantiate a generator object with a   seed and pass it around. The implicit global RandomState behind the numpy.random.* convenience functions can cause problems, especially when threads or other forms of concurrency are involved. Global state is always problematic. We categorically recommend avoiding using the convenience functions when reproducibility is involved._

Reply:
Ok, but what I am reading is *"pass it around"* and not re-instantiate in every function? Actually this passing around is probably what I meant by global, so not a global state.. but a global `rng` instance.